### PR TITLE
Bump minimum supported Swift version to 5.6

### PR DESCRIPTION
**Pull request by dfed**
This PR drops support for older Xcode and Swift versions. We're still in beta – let's not complicate our support story just yet.

### CODE REVIEW

File: Package.resolved [hunk 1]
```diff
@@ -1,61 +1,59 @@
 {
-  "object": {
```
Comment (dfed) [inline Package.resolved:2]: I committed the resolved file after opening with Xcode 13.3

File: Package@swift-5.5.swift [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.4
+// swift-tools-version:5.5
```
Comment (dfed) [inline Package@swift-5.5.swift:1]: Oops. This should always have been 5.5

File: Package@swift-5.6.swift [hunk 3]
```diff
@@ -0,0 +1,102 @@
+// swift-tools-version:5.6
```
Comment (dfed) [inline Package@swift-5.6.swift:1]: To make reviewing this file easier, here's the diff between this file and `Package@swift-5.5.swift`:

```
1c1
< // swift-tools-version:5.5
---
> // swift-tools-version:5.6
19c19
<     .package(name: "SwiftSyntax", url: "https://github.com/apple/swift-syntax.git", .exact("0.50500.0")),
---
>     .package(url: "https://github.com/apple/swift-syntax.git", exact: "0.50600.1"),
51c51,52
<         "SwiftSyntax",
---
>         .product(name: "SwiftSyntax", package: "swift-syntax"),
>         .product(name: "SwiftSyntaxParser", package: "swift-syntax"),
69c70,73
<       dependencies: ["SwiftSyntax"],
---
>       dependencies: [
>         .product(name: "SwiftSyntax", package: "swift-syntax"),
>         .product(name: "SwiftSyntaxParser", package: "swift-syntax"),
>       ],
83c87,89
<       dependencies: ["SwiftSyntax"],
---
>       dependencies: [
>         .product(name: "SwiftSyntax", package: "swift-syntax"),
>       ],
```

**Review by fdiaz — APPROVED** [review]

**Comment by dfed** [comment]
Force push rebased `main`. No changes.

**Comment by dfed** [comment]
I just pushed up a change to drop macOS support prior to v12, and Xcode support prior to v13. Trying to simplify while we're still in beta 😅

File: README.md [hunk 4]
```diff
@@ -14,7 +14,7 @@ This project is currently under development and can have breaking API changes.
 
 ## Requirements
 
-- Swift 5.4+
+- Swift 5.6+
```
Comment (fdiaz) [inline README.md:17]:
```suggestion
- Swift 5.6
```

No + anymore 😆

**Review by fdiaz — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (dfed) [reply README.md:17]: fair fair. we just gotta remember to update this once 6 comes out.

File: README.md [hunk 5]
```diff
@@ -14,7 +14,7 @@ This project is currently under development and can have breaking API changes.
 
 ## Requirements
 
-- Swift 5.4+
+- Swift 5.6
```
Comment (dfed) [inline README.md:17]:
```suggestion
- Swift 5.6
- Xcode 13.3+
```

**Comment by bachand** [comment]
Deferring to Francisco's review.

**Pull request merged**